src/go/guestagent/pkg/forwarder/serviceapi.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright © 2024 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package forwarder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/Masterminds/log-go"
+	"github.com/containers/gvisor-tap-vsock/pkg/types"
+)
+
+const (
+	exposeAPI   = "/services/forwarder/expose"
+	unexposeAPI = "/services/forwarder/unexpose"
+)
+
+var (
+	ErrAPI         = errors.New("error from API")
+	ErrExposeAPI   = fmt.Errorf("error from %s API", exposeAPI)
+	ErrUnexposeAPI = fmt.Errorf("error from %s API", unexposeAPI)
+)
+
+// APIForwarder forwards the PortMappings to /services/forwarder/expose
+// or /services/forwarder/unexpose that is host in the host-switch.
+type APIForwarder struct {
+	baseURL    string
+	httpClient http.Client
```

Comment:
```suggestion
	httpClient *http.Client
```
There's no need to make a copy of the client. (The fact that we needed to dereference `http.DefaultClient` in `NewAPIForwarder` clued me into this being probably the case.)